estate/models/estate_property.py

```diff
@@ -0,0 +1,120 @@
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare
+
+
+class EstateProperty(models.Model):
+    _name = 'estate.property'
+    _description = 'Estate property'
+    _order = 'id desc'
+
+    name = fields.Char('Title', required=True)
+    active = fields.Boolean('Active', default=True)
+    description = fields.Text()
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date(
+        'Available From',
+        copy=False,
+        default=lambda __: fields.Date.today() + relativedelta(months=3),
+    )
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer()
+    has_garage = fields.Boolean()
+    has_garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Orientation',
+        selection=[
+            ('north', 'North'),
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ],
+    )
+    state = fields.Selection(
+        string='State',
+        selection=[
+            ('new', 'New'),
+            ('received', 'Offer Received'),
+            ('accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        default='new',
+        required=True,
+        copy=False,
+    )
+    property_type_id = fields.Many2one('estate.property.type', string='Property Type')
+    salesman_id = fields.Many2one(
+        'res.users',
+        'Salesman',
+        default=lambda self: self.env.user,
+    )
+    buyer_id = fields.Many2one('res.partner', 'Buyer', copy=False)
+    tag_ids = fields.Many2many('estate.property.tag', string='Tags')
+    offer_ids = fields.One2many('estate.property.offer', 'property_id')
+    total_area = fields.Integer(compute='_compute_total')
+    best_price = fields.Float('Best Offer', compute='_compute_best_price')
+
+    _sql_constraints = [
+        (
+            'check_expected_price',
+            'CHECK(expected_price >= 0)',
+            'A property expected price must be strictly positive.',
```

Comment:
That's what _strictly_ means, no?
```suggestion
            'CHECK(expected_price > 0)',
            'A property expected price must be strictly positive.',
```